06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js

```diff
@@ -0,0 +1,276 @@
+   * @returns {Array} - An array consisiting of a sub array with each word
+   * and booleans to indicate their occurence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static constructIndex(words, wordsInEachBook, bookTitles) {
+    const checked = this
+    .checkOccurenceOfWords(words, wordsInEachBook);
+    const booksLength = bookTitles.length;
+    const booksChecked = this
+    .getWordOccurenceForEachBook(checked, booksLength);
+    const indexConstructed = this
+    .mapWordsWithOccurence(words, booksChecked);
+    return indexConstructed;
+  }
+
+  /**
+   * This method gets the title of each book in a file
+   * @static
+   * @param {Array} file - The content of the given file
+   * @returns {Array} - The Title of each book in the file
+   * @memberOf InvertedIndexUtility
+   */
+  static getTitlesOfEachBook(file) {
+    const titleList = [];
+    for (let i = 0; i < file.length; i += 1) {
+      titleList.push(file[i].title);
+    }
+    return titleList;
+  }
+
+  /**
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an Array that contains all the words in
+   * the book without duplication
+   */
+  static getAllWords(file) {
+    let words = '';
+    for (let i = 0; i < file.length; i += 1) {
+      words += `${file[i].text} `;
+    }
+    words = this.tokenize(words).slice(0, words.length - 1);
+    words = this.removeDuplicateWords(words);
+
+    return words.slice(0, words.length);
+  }
+
+  /**
+   * This function returns an array that consists of all the words in each book
+   * [[], [], ]
+   * @static
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an array that consist of all the words in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getAllWordsInEachBook(file) {
+    const doc = [];
+    let bookContent;
+    for (let i = 0; i < file.length; i += 1) {
+      bookContent = `${file[i].text} `;
+      bookContent = this.tokenize(bookContent);
+      bookContent = this.removeDuplicateWords(bookContent);
+      doc.push(bookContent.slice(0, bookContent.length));
+    }
+    return doc;
+  }
+
+  /**
+   * This function takes two arrays and checks
+   * if each of the words in the first array is present in the second array
+   * @static
+   * @param {Array} words - contains all the words
+   * in the book without duplication
+   * @param {Array} wordsInEachBook - contains all the  words
+   * in a given document
+   * @returns {Array} - An array that consist of
+   * booleans to indicate if the words are present in each book
+   * e.g [true, false, false, false, false, true . . .]
+   * @memberOf InvertedIndexUtility
+   */
+  static checkOccurenceOfWords(words, wordsInEachBook) {
+    const checked = [];
+
+    words.forEach((word) => {
+      for (let i = 0; i < wordsInEachBook.length; i += 1) {
```

Comment:
This right here is a violation of the Airbnb Style Guide. See [Link](https://github.com/airbnb/javascript#iterators--nope)